src/cattrs/gen.py

```diff
@@ -412,8 +412,10 @@ def make_iterable_unstructure_fn(cl: Any, converter, unstructure_to=None):
 
     fn_name = "unstructure_iterable"
 
-    # Let's try fishing out the type args.
-    if getattr(cl, "__args__", None) is not None:
+    # Let's try fishing out the type args
+    # Unspecified tuples have `__args__` as empty tuples, so guard
+    # against IndexError.
+    if getattr(cl, "__args__", None) not in (None, ()):
```

Comment:
Btw @Tinche, I could cherry-pick this commit and create a separate branch, because this is related to #209 .

That fix was not only for `t.List`, but also for `t.Set`, `t.FrozenSet` etc., but `t.Tuple` has, as I noted earlier, `t.Tuple.__args__ == ()`

Reply:
I wouldn't bother with another branch